core/services/relay/relay.go

```diff
@@ -28,6 +30,16 @@ var (
 	}
 )
 
+type LogPollerCapable interface {
+	RegisterLogFilters(args types.RelayArgs, q pg.Queryer) (err error)
+	UnregisterLogFilters(args types.RelayArgs, q pg.Queryer) (err error)
+}
+
+type JobHooks interface {
+	OnCreateJob(arg types.RelayArgs, q pg.Queryer) (err error)
+	OnDeleteJob(arg types.RelayArgs, q pg.Queryer) (err error)
```

Comment:
Are we absolutely stuck with an atomic transaction right now? We will have to give it up with LOOPs and it would simpler to leave it out from the start, especially since it is leaking in to the chain-agnostic API.

Reply:
Several months ago, we started exploring different potential solutions to the problems node ops were seeing with job tables getting out of sync while deleting and creating jobs due to things being non-atomic.  There were several attempts at fixes for it which didn't fix the problem, the only one we found that was bulletproof was to make everything atomic... so little by little over the past 4 or 5 months we've been working towards getting everything to be atomic.  This effort began before it became clear that the relay would be involved at all, so LOOP plugins were not taken into consideration.  It seems we'll have to create another spike to figure out an alternate solution, but I don't know of any easy solution.

Reply:
I suspect what we probably need is to implement some kind of reaper that runs as a service, and periodically cleans up any inconsistencies.  But that would need to be a separate project.